### DISCUSSION TITLE
ObjectPool.Get （10）与rabbitmq

### DISCUSSION THREAD

**Discussion by gwhzh21** [dotnetcore/FreeSql#1402] [Q&A]
当连不上rabbitmq的时候，在几百次RabbitMQ.Client.Exceptions.BrokerUnreachableException: None of the specified endpoints were reachable之后。System.Exception: ObjectPool.Get 获取超时（10秒）就会出现这个，但是他们不是两个东西吗？

**Comment by 2881099** [comment]
是不是本地套字节端口耗尽，试试此时 new Socket 连接任意远程会不会报错。

或者CPU和同步锁资源争抢，导致超过10秒未正常获取到链接。

或者此时监视打印 fsql.MasterPool.Statistics，这个在常见问题里有说明。

前两个是系统有关，最后这个监视打印是使用不当导致。

**Reply by gwhzh21** [reply]
收到。我尽量验证下。
